### PR TITLE
Add podcast show notes

### LINKED ISSUE #387: Add show notes to podcasts
For each episode of a podcast the administrator should be able to add show notes in CommonMark format.

- Permissions should align with if the person is allowed to upload episodes
- Show notes should be accessed for editing from `/Podcasts/EditEpisode/` 
- I don't believe we need these on add episode 
- Should be an "add show notes" button or "edit show notes" if they exist
- Show notes can probably be stored in a segment
- Segment should be forced to be named "Podcast - `podcast name` - Show notes #`episode number`"
- If the show notes are deleted then delete the segment entirely
- See approach for how to add seamless adding/editing of segments on the location details page (e.g. description segment)

### PR DESCRIPTION
**Pull request by madison-boman**
Fix #387 Add show notes to podcasts
- Permissions align with if the person is allowed to upload episodes
- Show notes are accessed for editing from `/Podcasts/EditEpisode/`
- Segment should be forced to be named "Podcast - `podcast name` - Show notes #`episode number`"
- Deleting delete the segment entirely
- Show notes are accessible from Promenade
- Promenade printing is improved so show notes should print well

Remaining steps:

- [x] Request translations
- [x] Add translations
- [x] Add permission for adding show notes
- [x] Add the migration and bump the version
- [x] Add Blockquote styling to be indented for transcript dialogue

### CODE REVIEW

File: src/Ops.Web/Areas/SiteManagement/Views/Podcasts/EpisodeDetails.cshtml [hunk 1]
```diff
@@ -196,4 +251,59 @@
             @Model.MaximumFileSizeMB MB
         </div>
     </div>
+    <div class="row form-group mt-4">
+        <div class="col-form-label text-md-right col-md-3">
+            <label asp-for="Episode.ShowNotesSegmentId">
+            </label>
+            <span class="fas fa-asterisk fa-xs d-inline-block ml-2 text-danger oc-required-field-marker">
```
Comment (iafb) [inline src/Ops.Web/Areas/SiteManagement/Views/Podcasts/EpisodeDetails.cshtml:258]: It seems like Show Notes aren't required so we might be able to remove this required asterisk, I might be wrong though.

File: src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs [hunk 2]
```diff
@@ -138,6 +142,12 @@ public async Task<IActionResult> Delete(IndexViewModel model)
         {
             try
             {
+                var episode = await _podcastService.GetEpisodeBySegmentIdAsync(model.Segment.Id);
+                if (episode != null)
+                {
+                    episode.ShowNotesSegmentId = null;
+                    await _podcastService.UpdatePodcastItemAsync(episode);
+                }
```
Comment (iafb) [inline src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs:150]: Show notes segment deletion should be moved to the Podcasts controller, users without SiteManagement permissions won't be able to scroll through the segment index to find the segment to delete it. Instead there should be a delete notes icon next to the edit notes icon when viewing an episodes details.

With this change podcast segments should be added to the `GetSegmentInUseByAsync` function in SegmentService so podcasts segments aren't accidentally deleted from the segment page.

File: src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs [hunk 3]
```diff
@@ -274,6 +284,19 @@ var pageLayoutId
                         "This segment is used for multiple locations: {0}",
                         string.Join(", ", locations.Select(_ => _.Name)));
                 }
+
+                var episode = await _podcastService.GetEpisodeBySegmentIdAsync(segment.Id);
```
Comment (iafb) [inline src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs:288]: A permission check for podcasts permission needs to be added to the `HasSegmentPermissionAsync` function in the SegmentsController, currently a user with just podcast permissions won't be able to view or edit the segment details.

**Review by iafb — CHANGES_REQUESTED** [review]

File: src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs (hunk 3 above)
Comment (madison-boman) [reply src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs:282]: Resolved in commit 577d9e3

File: src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs (hunk 2 above)
Comment (madison-boman) [reply src/Ops.Controllers/Areas/SiteManagement/SegmentsController.cs]: Resolved in commit 00bd5d1

**Comment by k7hpn** [comment]
This is looking good. To merge we need:

- [ ] Staff guidelines on adding notes
- [ ] Add the migration and bump the version

**Review by iafb — APPROVED** [review]

**Review by k7hpn — APPROVED** [review]

**Pull request merged**